tests/ubench/event_setup.cpp

```diff
@@ -0,0 +1,207 @@
+// Compare methods for performing the "event-setup" step for mc cell groups.
+// The key concern is how to take an unsorted set of events
+//
+// TODO: We assume that the cells in a cell group are numbered contiguously,
+// i.e. 0:ncells-1. The cells in an mc_cell_group are not typically thus,
+// instead a hash table is used to look up the cell_group local index from the
+// gid. A similar lookup should be added to theses tests, to moare accurately
+// reflect the mc_cell_group implementation.
+//
+// TODO: The staged_events output is a vector of postsynaptic_spike_event, not
+// a deliverable event.
+
+#include <random>
+#include <vector>
+
+#include <event_binner.hpp>
+#include <event_queue.hpp>
+#include <backends/event.hpp>
+
+#include <benchmark/benchmark.h>
+
+using namespace nest::mc;
+
+std::vector<postsynaptic_spike_event> generate_inputs(size_t ncells, size_t ev_per_cell) {
+    std::vector<postsynaptic_spike_event> input_events;
+    std::default_random_engine engine;
+    std::uniform_int_distribution<cell_gid_type>(0u, ncells);
+
+    std::default_random_engine gen;
```

Comment:
Let's not use `std::default_random_engine`: it's implementation defined, and could be particularly poor on one hand, and give a different sequence on different platforms on the other. If quality is not a particular concern, I'd stick with `std::minstd_rand`.

Reply:
using `std::mt19937`